internal/check/engine.go

```diff
@@ -66,7 +78,16 @@ func (e *Engine) subjectIsAllowed(ctx context.Context, requested *relationtuple.
 	return false, nil
 }
 
-func (e *Engine) checkOneIndirectionFurther(ctx context.Context, requested *relationtuple.InternalRelationTuple, expandQuery *relationtuple.RelationQuery) (bool, error) {
+func (e *Engine) checkOneIndirectionFurther(
+	ctx context.Context,
+	requested *relationtuple.InternalRelationTuple,
+	expandQuery *relationtuple.RelationQuery,
+	restDepth int,
+) (bool, error) {
+	if restDepth <= 0 {
+		return false, nil
```

Comment:
For debugging purposes, it would be helpful to log here that the limit was reached.
```suggestion
		e.d.Logger().WithFields((*relationtuple.InternalRelationTuple)(expandQuery).ToLoggerFields()).Debug("reached max-depth, therefore this query will not be further expanded")
		return false, nil
```

Reply:
Type conversion won't work here since `RelationQuery` isn't a subset of `InternalRelationTuple`